python/cudf/cudf/core/reshape.py

```diff
@@ -278,9 +327,25 @@ def concat(objs, axis=0, ignore_index=False, sort=None):
                 result = objs[0].copy()
             return result
         else:
-            return cudf.DataFrame._concat(
-                objs, axis=axis, ignore_index=ignore_index, sort=sort
+            result = cudf.DataFrame._concat(
+                objs,
+                axis=axis,
+                join=join,
+                ignore_index=ignore_index,
+                sort=sort,
             )
+            if ignore_index and join == "inner" and result.empty:
+                # this occurs with no overlapping column names
+                result.index = cudf.RangeIndex(sum(len(obj) for obj in objs))
```

Comment:
Here too we should be deciding the `index` based on `ignore_index` and not based on `join` parameter. This block of logic from 330 to 348 can be simplified, when we are passing in `ignore_index` to `DataFrame._concat` we shouldn't again be handling it here outside as a special case.